wagtail/embeds/models.py

```diff
@@ -22,22 +22,23 @@ class Embed(models.Model):
     If an instance of this model is deleted, it will be automatically refetched
     next time the embed code is needed.
     """
-    url = models.URLField()
+
+    url = models.TextField()
     max_width = models.SmallIntegerField(null=True, blank=True)
+    hash = models.CharField(max_length=32, unique=True, db_index=True)
     type = models.CharField(max_length=10, choices=EMBED_TYPES)
     html = models.TextField(blank=True)
     title = models.TextField(blank=True)
     author_name = models.TextField(blank=True)
     provider_name = models.TextField(blank=True)
-    thumbnail_url = models.URLField(max_length=255, null=True, blank=True)
+    thumbnail_url = models.TextField(blank=True)
```

Comment:
@allcaps Just noticed we have lost the `null=True` here. This makes sense from a consistency point of view, but we'll want to check whether the ORM is checking for null values anywhere before merging.

Reply:
The Django docs mentions not add `null=True` to string based fields:

> Avoid using null on string-based fields such as CharField and TextField. If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string. In most cases, it’s redundant to have two possible values for “no data;” the Django convention is to use the empty string, not NULL. One exception is when a CharField has both unique=True and blank=True set. In this situation, null=True is required to avoid unique constraint violations when saving multiple objects with blank values.
https://docs.djangoproject.com/en/3.1/ref/models/fields/#null

Adding `null=True` on string based fields is a common mistake. I can't see any good reason because `thumbnail_url` does not have unique constraints.

Our migration already has `default=""`, AFAIK any `Null` values will be converted to empty strings.

Is searched `wagtail/wagtail/embeds` module for `isnull`. Zero matches.
I also searched for `None`, 72 matches, as far as I can see, not related to `thumbnail_url`.
@ababic Will you also search the code whether the ORM is checking for null values?

Reply:
@allcaps yep, I'll have a quick look. I imagine all will be fine! I'm totally with you on the reasoning behind the change - it just usually pays to be thorough where field changes are concerned.

Reply:
@allcaps LGTM :)